File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.py
def solution(arr):
    answer = []
    a = []
    for i in range(len(arr)):
        if len(answer) == 0 or arr[i] != answer[-1]:
            answer.append(arr[i])
        elif arr[i] == answer[-1]:
            answer.pop()
    if len(answer) == 0 :
        answer = [-1]
    return answer

File: 프로그래머스/unrated/181859. 배열 만들기 6/배열 만들기 6.js
function solution(arr) {
    var answer = [];
    let i = 0;
    while(i != arr.length){
        if(answer.length == 0){
            answer.push(arr[i])
        }else if(answer.slice(-1) == arr[i]){
            answer.pop()
        }else if(answer.slice(-1) != arr[i]){
            answer.push(arr[i])
        }
        i++;
    }
    if (answer.length === 0){
        return answer = [-1]
    }
    return answer;
}